frame/erc1155/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+		PalletId,
+	};
+
+	pub type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+	pub type AccountIdLookupOf<T> = <<T as frame_system::Config>::Lookup as StaticLookup>::Source;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config:
+		frame_system::Config + pallet_uniques::Config + pallet_assets::Config
+	{
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		type Currency: Currency<Self::AccountId>;
+
+		type CollectionId;
+
+		type ItemId;
+
+		type AssetId;
+
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn assets_minted)]
+	// TODO: query amount minted from pallet assets instead of storing it locally.
+	// Add a public getter function to pallet assets. 
+	pub type AssetsMinted<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		<T as pallet_assets::Config>::AssetId,
+		BalanceOf<T>,
+		OptionQuery,
+	>;
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		NFTLocked(
+			<T as pallet_uniques::Config>::CollectionId,
+			<T as pallet_uniques::Config>::ItemId,
+		),
+		AssetCreated(<T as pallet_assets::Config>::AssetId),
+		AssetMinted(<T as pallet_assets::Config>::AssetId, <T as pallet_assets::Config>::Balance),
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AssetAlreadyRegistered,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(2).ref_time())]
+		pub fn lock_nft_create_asset(
+			origin: OriginFor<T>,
+			collection_id: <T as pallet_uniques::Config>::CollectionId,
+			item_id: <T as pallet_uniques::Config>::ItemId,
+			asset_id: <T as pallet_assets::Config>::AssetId,
+			beneficiary: AccountIdLookupOf<T>,
+			min_balance: <T as pallet_assets::Config>::Balance,
+			amount: <T as pallet_assets::Config>::Balance,
+		) -> DispatchResult {
+			let _who = ensure_signed(origin.clone())?;
+			let admin_account_id = Self::pallet_account_id();
+			let admin = T::Lookup::unlookup(admin_account_id.clone());
+
+			match Self::do_lock_nft(origin.clone(), collection_id, item_id) {
+				Err(e) => return Err(e),
+				Ok(()) => match Self::do_create_asset(origin.clone(), asset_id, admin, min_balance)
+				{
+					Err(e) => return Err(e),
+					Ok(()) => match Self::do_mint_asset(
+						// Minting the asset is only possible from the pallet's origin.
+						// TODO: should the minting be possible from the owner's account?
+						frame_system::RawOrigin::Signed(admin_account_id).into(),
+						asset_id,
+						beneficiary,
+						amount,
+					) {
+						Err(e) => return Err(e),
+						Ok(()) => {
+							Self::deposit_event(Event::NFTLocked(collection_id, item_id));
+							Self::deposit_event(Event::AssetCreated(asset_id));
+							Self::deposit_event(Event::AssetMinted(asset_id, amount));
```

Comment:
assets pallet will emit those events, no need to have them here

Reply:
Fair enough, will remove these